c++/include/orc/Reader.hh

```diff
@@ -149,6 +149,25 @@ namespace orc {
      */
     RowReaderOptions& includeTypes(const std::list<uint64_t>& types);
 
+    /**
+     * A map type of <typeId, ReadIntent>.
+     */
+    typedef std::map<uint64_t, ReadIntent> TypeReadIntents;
+
+    /**
+     * Selects which type ids to read and specific ReadIntents for each
+     * type id. The root type is always 0 and the rest of the types are
+     * labeled in a preorder traversal of the tree. The parent types are
```

Comment:
`parent` -> `ancestor`? `parent` means only the direct nodes technically.

Reply:
Done.